src/lib/math/pcurves/pcurves_impl/pcurves_impl.h

```diff
@@ -530,6 +530,13 @@ class AffineCurvePoint {
       FieldElement m_y;
 };
 
+template <typename FE>
+const FE& inv_2() {
+   // TODO could be a constexpr, but some compilers get cranky
+   static const FE INV_2 = FE::from_word(2).invert();
+   return INV_2;
+}
```

Comment:
😞 

Perhaps it's worth allow-listing the compilers that are happy to do this as consteval?

Like so:

```C++
#if GOOD_BOI_COMPILER
template <typename FE>
consteval FE inv_2() {
   return FE::from_word(2).invert();
}
#else
// ... the runtime lazy-evaluation
#endif
```

Reply:
Unfortunately it seems to be not just compiler type to type but for example GCC 14 on i386 bad, GCC 14 on x86-64 OK ... I think the best we could do would be doing constexpr on specific known compilers/versions that can handle it. Which might still be worthwhile

Reply:
Aha just figured out a nice trick that makes this fast for constexpr - if P is prime then x=(P/2)+1 is the inverse of 2 modulo P